java-checks/src/test/files/checks/LoopsOnSameSetCheck.java

```diff
@@ -7,17 +7,22 @@
     return null;
   }
 
+  private List<String> globalStrings;
+
   public void doSomethingToAList(List<String> strings, List<String> strings2) {
     for (String str : strings);
-    for (String str : strings); // Noncompliant {{Combine this loop with the one that starts on line 11.}}
+    foo();
+    for (String str : strings); // Noncompliant {{Combine this loop with the one that starts on line 13.}}
     strings = null;
     for (String str : strings); // strings get reassigned
     for (String str : getList());
-    for (String str : getList()); // Noncompliant {{Combine this loop with the one that starts on line 15.}}
+    for (String str : getList()); // Noncompliant {{Combine this loop with the one that starts on line 18.}}
+    foo();
+    for (String str : getList()); // no issue if something in between for loops
```

Comment:
@m-g-sonar nope, when the iterable is an identifier which is not a field there can be things in between the for loops as long as they don't reference it.